include/hpp/constraints/matrix-view.hh

```diff
@@ -873,12 +873,16 @@ class MatrixBlockView
     }
     // ++it
     block_iterator& operator++() {
-      _ro.setValue(_ro.value() + rs());
+      if (!AllRows) {
+        _ro.setValue(_ro.value() + rs());
+      }
       ++row;
       if (row == (size_type)view.m_rows.size()) {
         row = 0;
         _ro.setValue(0);
-        _co.setValue(_co.value() + cs());
+        if (!AllCols) {
+          _co.setValue(_co.value() + cs());
+        }
         ++col;
```

Comment:
When `AllCols` is true, I expect that the iterator becomes invalid when reaching this code.
```
// All the blocks have been iterated on.
assert(!valid());
```